raphtory-graphql/src/data.rs

```diff
@@ -85,94 +102,830 @@ impl Data {
+    if !path.exists() {
+        return Ok(None);
+    }
+    if path.is_dir() {
+        if is_disk_graph_dir(path) {
+            get_disk_graph_from_path(path)
+        } else {
+            Ok(None)
+        }
+    } else {
+        let (name, graph) = load_bincode_graph(work_dir, path)?;
+        println!("Graph loaded = {}", path.display());
+        Ok(Some((name, IndexedGraph::from_graph(&graph.into())?)))
+    }
+}
 
-        fn add_to_graphs(
-            graphs: &mut HashMap<String, IndexedGraph<MaterializedGraph>>,
-            graph_name: &str,
-            graph: &MaterializedGraph,
-        ) {
-            if let Some(old_graph) = graphs.insert(
-                graph_name.to_string(),
-                IndexedGraph::from_graph(graph).expect("Unable to index graph"),
-            ) {
-                let old_path = old_graph.properties().get("path").unwrap_str();
-                let name = old_graph.properties().get("name").unwrap_str();
-                panic!(
-                    "Graph with name {} defined multiple times, first file: {}, second file: {}",
-                    name, old_path, graph_name
-                );
+pub(crate) fn get_graphs_from_work_dir(
+    work_dir: &Path,
+) -> Result<HashMap<String, IndexedGraph<MaterializedGraph>>> {
+    fn get_graph_paths(work_dir: &Path) -> Vec<PathBuf> {
+        let mut paths = Vec::new();
+        if let Ok(entries) = fs::read_dir(work_dir) {
+            for entry in entries {
+                if let Ok(entry) = entry {
+                    paths.push(entry.path());
+                }
             }
         }
+        paths
+    }
+
+    let mut graphs = HashMap::new();
+    for path in get_graph_paths(work_dir) {
+        if let Some((name, graph)) = get_graph_from_path(work_dir, &path)? {
+            graphs.insert(name, graph);
+        }
+    }
+
+    Ok(graphs)
+}
+
+fn is_disk_graph_dir(path: &Path) -> bool {
+    // Check if the directory contains files specific to disk_graph graphs
+    let files = fs::read_dir(path).unwrap();
+    let mut has_disk_graph_files = false;
+    for file in files {
+        let file_name = file.unwrap().file_name().into_string().unwrap();
+        if file_name.ends_with(".ipc") {
+            has_disk_graph_files = true;
+            break;
+        }
+    }
+    has_disk_graph_files
+}
+
+fn get_graph_name(path: &Path, graph: &MaterializedGraph) -> String {
+    graph
+        .properties()
+        .get("name")
+        .into_str()
+        .map(|v| v.to_string())
+        .unwrap_or_else(|| path.file_name().unwrap().to_str().unwrap().to_owned())
+}
+
+fn save_graphs_to_work_dir(
+    work_dir: &Path,
+    graphs: &HashMap<String, MaterializedGraph>,
+) -> Result<()> {
+    for (name, graph) in graphs {
+        let path = work_dir.join(&name);
+        graph.save_to_path(&path)?;
+    }
+    Ok(())
+}
+
+fn load_bincode_graph(work_dir: &Path, path: &Path) -> Result<(String, MaterializedGraph)> {
```

Comment:
assuming we want these formats to be stable we should use the new StableEncoder so our graphs can be restored even if the structure of the graph changes